examples/credel.c

```diff
@@ -0,0 +1,477 @@
+
+    if (ds < 0)
+        errx(EXIT_FAILURE, "H5Dcreate(, \"%s\", ) failed", dname);
+
+    if (s->verbose > 0)
+        fprintf(stderr, "Created dataset %s\n", dname);
+    s->dataset[didx % ndatasets] = ds;
+    s->stats.datasets.created++;
+}
+
+static void
+create_and_write_dataset(state_t *s, const int64_t didx)
+{
+#if 0
+	const int64_t gidx = didx / 2;
+	const int ngroups = __arraycount(s->group);
+	const hid_t g = s->group[gidx % ngroups];
+
+	if (H5Odisable_mdc_flushes(g) < 0)
+		err(EXIT_FAILURE, "H5Odisable_mdc_flushes failed");
+#endif
+
+    create_dataset(s, didx);
+    write_dataset(s, didx);
+
+#if 0
+	if (H5Oenable_mdc_flushes(g) < 0)
+		err(EXIT_FAILURE, "H5Oenable_mdc_flushes failed");
+#endif
+}
+
+static void
+handle_signal(int signo)
+{
+    char msg[] = "Handling signal\n";
+    write(STDERR_FILENO, msg, sizeof(msg) - 1);
+    unbroken = 0;
+}
+
+static void
+disestablish_handler(const struct sigaction *osa)
+{
+    if (sigaction(SIGINT, osa, NULL) == -1)
+        err(EXIT_FAILURE, "%s: sigaction", __func__);
+}
+
+static void
+establish_handler(struct sigaction *osa)
+{
+    struct sigaction sa;
+
+    memset(&sa, '\0', sizeof(sa));
+    sa.sa_handler = handle_signal;
+    sigemptyset(&sa.sa_mask);
+    if (sigaction(SIGINT, &sa, osa) == -1)
+        err(EXIT_FAILURE, "%s: sigaction", __func__);
+}
+
+int
+main(int argc, char **argv)
+{
+    hid_t                 fapl, fcpl;
+    struct sigaction      osa;
+    state_t               storage;
+    state_t *             s = &storage;
+    int64_t               i;
+    H5F_vfd_swmr_config_t config;
+
+    memset(&config, '\0', sizeof(config));
+
+    state_init(s, argc, argv);
+
+    fapl = H5Pcreate(H5P_FILE_ACCESS);
+    if (fapl < 0) {
+        errx(EXIT_FAILURE, "%s.%d H5Pcreate failed", __func__, __LINE__);
+    }
+
+    fcpl = H5Pcreate(H5P_FILE_CREATE);
+    if (fcpl < 0) {
+        errx(EXIT_FAILURE, "%s.%d H5Pcreate failed", __func__, __LINE__);
+    }
+
+    config.version  = H5F__CURR_VFD_SWMR_CONFIG_VERSION;
+    config.tick_len = 4;
+    config.max_lag  = 5;
+#if 0 /* raw-data flushing is not implemented */
+	config.flush_raw_data = true;
+#endif
+    config.writer            = true;
+    config.md_pages_reserved = 128;
+    strlcpy(config.md_file_path, "./my_md_file", sizeof(config.md_file_path));
+
+    /* Enable page buffering */
+    if (H5Pset_page_buffer_size(fapl, 4096, 100, 0) < 0)
+        errx(EXIT_FAILURE, "H5Pset_page_buffer_size failed");
+
+    /* Enable VFD SWMR configuration */
+    if (s->use_vfd_swmr && H5Pset_vfd_swmr_config(fapl, &config) < 0)
+        errx(EXIT_FAILURE, "H5Pset_vfd_swmr_config failed");
+
+    /* Set file space strategy to paged aggregation in fcpl.
+     * Page buffering *requires* this strategy.
+     *
+     * I set the free-space threshold to 1GB so that deleted
+     * datasets are not recycled.
+     */
+    if (H5Pset_file_space_strategy(fcpl, H5F_FSPACE_STRATEGY_PAGE, false, 1024 * 1024 * 1024) < 0)
```

Comment:
There's a lot of "fiddly bits" to getting these three properties set up, and all appear to be necessary.  It would be nice if there was a high-level routines (H5Psetup_vfd_s_swmr, maybe?) which took the fapl & fcpl, and the appropriate parameters for H5Pset_page_buffer, H5Pset_vfd_swmr_config and H5Pset_file_space_strategy and did all the setup for the user.  Would likely cut down considerably on user pain and support costs for the feature.